src/iris/bin/sender.py

```diff
@@ -1164,12 +1327,34 @@ def mock_gwatch_renewer():
 
 
 def sender_shutdown():
-    logger.info('Shutting server..')
+    global shutdown_started
+
+    # Make control+c or some other kill signal force quit the second time it happens
+    if shutdown_started:
+        logger.info('Force exiting')
+        os._exit(0)
+    else:
+        shutdown_started = True
+        logger.info('Shutting server..')
 
     # Immediately release all locks and give up any master status and slave presence
     if coordinator:
         coordinator.leave_cluster()
 
+    # Try to wait for all sender greenlets to die. Get a list, instead of generator,
+    # of all task objects so we can iterate over it multiple times.
+    all_worker_tasks = list(itertools.chain.from_iterable(
```

Comment:
Not sure what the goal of this line is - would just list concatenation not suffice here?

Reply:
The values themselves are lists. Using itertools to flatten them.